### PR TITLE
Update Dagger and build.

### PR DESCRIPTION
**Pull request by SuperQ**
* Update dagger actions to use pinnning.
* Add /dagger to dependabot to keep the Dagger modules up-to-date.
* Update dagger default Go version.
* Update tutorials/whatsup Go version.
* Update actions to use pinning.

### CODE REVIEW

**Comment by SuperQ** [comment]
Another yak shaving for #1601

File: .github/workflows/dagger-golangci-lint.yml [hunk 1]
```diff
@@ -17,15 +17,15 @@ permissions:  # added using https://github.com/step-security/secure-repo
 jobs:
   golangci:
     permissions:
-      contents: read  # for actions/checkout to fetch code
+      contents: read  # for actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```
Comment (ArthurSens) [inline .github/workflows/dagger-golangci-lint.yml:20]: Was this intentional? This is a comment, so I'm not sure if dependabot will keep this up-to-date as well 🤔

**Review by ArthurSens — APPROVED** [review]
I just have one question, but code LGTM

Thanks for working on this :)

File: .github/workflows/dagger-golangci-lint.yml (hunk 1 above)
Comment (SuperQ) [reply .github/workflows/dagger-golangci-lint.yml:20]: No, this is a `sed` skill issue on my part. 😹

Comment (ArthurSens) [inline .github/workflows/dagger-golangci-lint.yml:20]:
```suggestion
      contents: read  # for actions/checkout to fetch code
```

Comment (ArthurSens) [reply .github/workflows/dagger-golangci-lint.yml]: Added a commit reverting it then :)

**Pull request merged**